File: app/src/main/java/com/lightsoft/parceler/EmployeeDetailsActivity.java
package com.lightsoft.parceler;

import android.app.Activity;
import android.content.Intent;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.view.View;
import android.widget.TextView;

import org.parceler.Parcels;

public class EmployeeDetailsActivity extends Activity {

    public static final String ARG_EMPLOYEE = "employee";

    public static final int EDIT_REQ_CODE = 101;

    private Employee employee;

    private TextView textName;

    private TextView textLastName;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_employee_details);
        if (savedInstanceState != null && savedInstanceState.containsKey(ARG_EMPLOYEE))
            employee = Parcels.unwrap(savedInstanceState.getParcelable(ARG_EMPLOYEE));
        else
            employee = new Employee("John", "Doe");

        textName = findViewById(R.id.textName);
        textLastName = findViewById(R.id.textLastName);

        initializeViews();
    }

    private void initializeViews() {
        textName.setText(employee.getName());
        textLastName.setText(employee.getLastName());
    }

    @Override
    protected void onSaveInstanceState(Bundle outState) {
        Bundle bundle = outState != null ? outState : new Bundle();
        bundle.putParcelable(ARG_EMPLOYEE, Parcels.wrap(employee));
        super.onSaveInstanceState(bundle);
    }

    public void onEditClicked(View view) {
        Intent intent = new Intent(this, EditEmployeeActivity.class);
        intent.putExtra(ARG_EMPLOYEE, Parcels.wrap(employee));
        startActivityForResult(intent, EDIT_REQ_CODE);
    }

    @Override
    protected void onActivityResult(int requestCode, int resultCode, Intent data) {
        super.onActivityResult(requestCode, resultCode, data);
        if (requestCode == EDIT_REQ_CODE && resultCode == RESULT_OK && data != null) {
            employee = Parcels.unwrap(data.getParcelableExtra(ARG_EMPLOYEE));
            initializeViews();
        }
    }
}

File: kotlinparceler/src/main/java/com/lightsoft/kotlinparceler/EmployeeDetailsActivity.kt
package com.lightsoft.kotlinparceler

import android.content.Intent
import android.os.Bundle
import android.view.View
import com.lightsoft.kotlinparceler.model.Employee
import kotlinx.android.synthetic.main.activity_employee_details.*
import org.parceler.Parcels

class EmployeeDetailsActivity : android.app.Activity() {

    private var employee: Employee? = null


    override fun onCreate(savedInstanceState: android.os.Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_employee_details)
        employee = Parcels.unwrap(savedInstanceState?.getParcelable(ARG_EMPLOYEE)) ?: Employee("John", "Doe", null)
        initializeViews()
    }

    private fun initializeViews() {
        textName.text = employee?.name
        textLastName.text = employee?.lastName
    }


    override fun onSaveInstanceState(outState: android.os.Bundle?) {
        val bundle = outState ?: Bundle()
        bundle.putParcelable(ARG_EMPLOYEE, Parcels.wrap(employee))
        super.onSaveInstanceState(bundle)
    }

    fun onEditClicked(@Suppress("UNUSED_PARAMETER") view: View) {
        val intent = Intent(this, EditEmployeeActivity::class.java)
        intent.putExtra(ARG_EMPLOYEE, Parcels.wrap(employee))
        startActivityForResult(intent, EDIT_REQ_CODE)
    }

    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
        super.onActivityResult(requestCode, resultCode, data)
        if (requestCode == EDIT_REQ_CODE && resultCode == RESULT_OK) {
            employee = Parcels.unwrap(data?.getParcelableExtra(ARG_EMPLOYEE))
            initializeViews()
        }
    }
}